cfn-resources/integration-tests/project/project_test.go

```diff
@@ -0,0 +1,237 @@
+		testUpdateStack(t, testCtx)
+	})
+
+	t.Run("Delete Stack", func(t *testing.T) {
+		testDeleteStack(t, testCtx)
+	})
+}
+
+func setupSuite(t *testing.T) *LocalTestContext {
+	t.Helper()
+	t.Log("Setting up suite")
+	testCtx := new(LocalTestContext)
+	testCtx.setUp(t)
+
+	return testCtx
+}
+
+func (c *LocalTestContext) setUp(t *testing.T) {
+	c.resourceCtx = util.InitResourceCtx(e2eRandSuffix, resourceTypeName, resourceDirectory)
+	c.setClients(t)
+	util.PublishToPrivateRegistry(t, c.resourceCtx)
+	c.setupPrerequisites(t)
+}
+
+func testIsTemplateValid(t *testing.T, c *LocalTestContext) {
+	t.Helper()
+
+	isValid := util.ValidateTemplate(t, c.cfnClient, c.template)
+	a := assert.New(t)
+	a.True(isValid)
+}
+
+func testCreateStack(t *testing.T, c *LocalTestContext) {
+	t.Helper()
+
+	output := util.CreateStack(t, c.cfnClient, stackName, c.template)
+	c.projectTmplObj.ProjectID = getProjectIDFromStack(output)
+
+	project, getProjectResponse, err := c.atlasClient.Projects.GetOneProject(ctx.Background(), c.projectTmplObj.ProjectID)
+	util.FailNowIfError(t, "Error while retrieving Project from Atlas: %v", err)
+
+	teamsAssigned, _, _ := c.atlasClient.Projects.GetProjectTeamsAssigned(ctx.Background(), project.ID)
+	util.FailNowIfError(t, "Error when retrieving Project TeamsAssigned: %v", err)
+
+	a := assert.New(t)
+	a.Equal(c.projectTmplObj.TeamID, teamsAssigned.Results[0].TeamID)
+	a.Equal(200, getProjectResponse.StatusCode)
+}
+
+func testUpdateStack(t *testing.T, c *LocalTestContext) {
+	t.Helper()
+
+	// create CFN template with updated project name
+	c.projectTmplObj.Name += "-updated"
+	c.template, c.err = newCFNTemplate(c.projectTmplObj)
+
+	output := util.UpdateStack(t, c.cfnClient, stackName, c.template)
+	c.projectTmplObj.ProjectID = getProjectIDFromStack(output)
+
+	project, _, err := c.atlasClient.Projects.GetOneProject(ctx.Background(), c.projectTmplObj.ProjectID)
+	util.FailNowIfError(t, "Error while retrieving Project from Atlas: %v", err)
+
+	teamsAssigned, _, err := c.atlasClient.Projects.GetProjectTeamsAssigned(ctx.Background(), project.ID)
+	util.FailNowIfError(t, "Error when retrieving Project TeamsAssigned: %v", err)
+
+	a := assert.New(t)
+	a.Equal(c.projectTmplObj.TeamID, teamsAssigned.Results[0].TeamID)
+	a.Equal(project.Name, c.projectTmplObj.Name)
+}
+
+func testDeleteStack(t *testing.T, c *LocalTestContext) {
+	t.Helper()
+
+	util.DeleteStack(t, c.cfnClient, stackName)
+
+	_, resp, _ := c.atlasClient.Projects.GetOneProject(ctx.Background(), c.projectTmplObj.ProjectID)
+
+	a := assert.New(t)
+	a.Equal(resp.StatusCode, 404)
+}
+
+func (c *LocalTestContext) setClients(t *testing.T) {
+	t.Helper()
+
+	t.Log("Setting clients")
+	c.atlasClient, c.err = util.NewMongoDBClient()
+	util.FailNowIfError(t, "Unable to create atlas client: %v", c.err)
+
+	c.cfnClient, c.err = util.NewCFNClient()
+	util.FailNowIfError(t, "Unable to create AWS client, please check AWS config is correctly setup: %v", c.err)
+}
```

Comment:
Shouldn't this function be in the `atlas_helper` file so that it can be used by other tests?

Reply:
will update